content/smartmodules/certified/json-sql.md

```diff
@@ -0,0 +1,164 @@
+---
+title: JSON to SQL Mapping
+menu: Json-Sql
+weight: 20
+---
+
+This is certified by InfinyOn [map-type]({{<ref "../transform/map.md" >}}) SmartModule that converts arbitrary JSON records into [SQL model](https://github.com/infinyon/fluvio-connectors/blob/main/rust-connectors/models/fluvio-model-sql), which is a self-describing representation of SQL INSERT statements. Usually used in pipelines with [SQL Sink Connector][sql-sink-connector], the latter executes statements from the model against configured SQL database.
+
+### Mapping
+The mapping between incoming JSON records and the resulting SQL record is defined in the configuration of the SmartModule in the `mapping` init parameter. Let's look at the example:
+
+%copy%
+```yaml
+- uses: infinyon/json-sql@0.1.0
+  with:
+    mapping:
+      table: "target_table"
+      map-columns:
+        "device_id":
+          json-key: "device.device_id"
+          value:
+            type: "int"
+            required: true
+        "device_type":
+          json-key: "device.type"
+          value:
+            type: "text"
+            default: "mobile"
+        "record":
+          json-key: "$"
+          value:
+            type: "jsonb"
+            required: true
+```
+
+Here, we create insert statements to `target_table` database table. Each statement has three columns:
+* **device_id** with `int` type.  The value for this column will be sought under path `$.device.device_id` in the input record. If it is not present, the error will be thrown, as the mapping states, it is a required field.
```

Comment:
~~be sought under path `$.device.device_id` in the input record~~ -> looked-up at the following hierarchy `$.device.device_id` of the input record.